### PR TITLE
feat(core): intercept OSC 133/7 shell-integration marks (TREK-164)

### PR DESCRIPTION
**Pull request by jbabin91**
## Summary

Wires OSC 133 (shell-integration semantic marks) and OSC 7 (working directory) into the terminal, completing the last remaining task in EPIC-5 (Phase 0.4). The data layer for these marks already existed (`Row.semantic_mark`/`mark_metadata`, wire + archive serialization, `find_prompt_in_buffer`) but was always `None` in production because nothing set the marks.

**Root cause:** vte 0.15.0 (verified latest — no newer release) has no `ansi::Handler` callback for OSC 133/7; its internal `osc_dispatch` routes both to `unhandled()`. Confirmed against the vendored vte source. So we intercept rather than rely on a handler hook (Scope option 3).

## Approach

New `ShellIntegrationScanner` (`crates/oakterm-terminal/src/shell_integration.rs`) runs alongside the VT processor at the `ScreenSet::process_bytes` boundary:

- Byte-level OSC state machine that finds complete OSC 7/133 sequences (BEL or ST terminated), **persisting partial-sequence state across PTY read chunks**.
- Honors CAN/SUB cancellation and caps payloads at `MAX_OSC` (4096).
- Splits each read at the OSC terminator so a mark attaches to the row the cursor occupies **after prior bytes render** — every byte still reaches vte unchanged.

Decoding → row mutations:

| Sequence | `SemanticMark` | `MarkMetadata` |
| --- | --- | --- |
| `133;A` / `B` / `C` | PromptStart / InputStart / OutputStart | — |
| `133;D[;<exit>]` | OutputEnd | `ExitCode` if present |
| `7;file://host/path` | — | `WorkingDirectory` (percent-decoded; host dropped) |

Correctly disambiguates `7` vs `70` and `133` vs `1337`.

## Testing

- 29 scanner unit tests (chunk splits, BEL/ST terminators, CAN/SUB, overflow recovery, percent-decode edge cases, lossy UTF-8, code disambiguation).
- 6 `ScreenSet` integration tests (cursor-row timing, exit-code metadata, cwd, cross-read-chunk split, dirty seqno, stale-metadata clearing).
- End-to-end daemon test: `OSC 133;A` → scrollback → `find_prompt_in_buffer`.
- Benchmarks added for the every-byte hot path: `screenset_plain_ascii` (~111 MiB/s, above the 100 MB/s ground-state target) and `screenset_shell_session`.
- `mise run check` (fmt + clippy-pedantic + md-lint) and full `mise run test` green.

## Review

Ran the full multi-agent review cycle (Codex + code/test/silent-failure/type-design + maintainability + spec-conformance), 2 iterations. Fixes applied inline: CAN/SUB cancellation, stale-metadata clearing on bare `133;D`, logging breadcrumbs on cwd drop paths, bounds-check ordering. Spec-conformance verdict: faithful to ADR-0008 Phase 0, data-layer boundary respected.

## Follow-ups (tracked, out of scope here)

- **TREK-165** — `Row` multi-mark data model (a row can only hold one mark / one metadata slot today; needed for Phase 1 command-blocks per ADR-0015).
- **TREK-134** (CMT-111) — OSC 7 cwd now lives as row metadata; hostname is dropped.

Closes TREK-164.

https://claude.ai/code/session_013Lw1gM3artqQ2MStk5UhwA

### CODE REVIEW

File: crates/oakterm-terminal/src/grid/mod.rs [hunk 1]
```diff
@@ -303,16 +307,60 @@ impl ScreenSet {
...
+            ShellMark::CommandFinished(exit) => {
+                line.semantic_mark = SemanticMark::OutputEnd;
+                // Own this row's metadata: set the exit code, or clear any stale
+                // metadata (e.g. a prior cwd) when the D mark carries no code.
+                line.mark_metadata = exit.map(MarkMetadata::ExitCode);
+            }
+            ShellMark::WorkingDirectory(path) => {
+                line.mark_metadata = Some(MarkMetadata::WorkingDirectory(path.clone()));
+            }
+        }
```
[the remark this answers is not in the document]
Comment (jbabin91) [reply crates/oakterm-terminal/src/grid/mod.rs:360]: Good catch — fixed in 1944572. This is the mirror case of a stale-metadata issue Codex flagged (bare `133;D` not clearing a prior cwd). Rather than special-casing each mark, `apply_shell_mark` now enforces the invariant that an `ExitCode` may only coexist with an `OutputEnd` mark: after applying any mark, a lingering `ExitCode` on a non-`OutputEnd` row is cleared, while a `WorkingDirectory` (OSC 7) still survives a prompt mark on the same row. Both cases are covered by a new test (`osc_133_prompt_sheds_stale_exit_code_but_keeps_cwd`).

The deeper cause is that a `Row` has a single `mark_metadata` slot; the general fix (multiple marks per row) is tracked as TREK-165.
